discovery/gossiper.go

```diff
@@ -1072,39 +1151,23 @@ func (d *AuthenticatedGossiper) networkHandler() {
 				continue
 			}
 
-			// For the set of peers that have an active gossip
-			// syncers, we'll collect their pubkeys so we can avoid
-			// sending them the full message blast below.
-			syncerPeers := d.syncMgr.GossipSyncers()
-
-			log.Infof("Broadcasting batch of %v new announcements",
+			subBatchSize := calculateSubBatchSize(
+				d.cfg.TrickleDelay,
+				d.cfg.SubBatchDelaySecond,
 				len(announcementBatch))
 
-			// We'll first attempt to filter out this new message
-			// for all peers that have active gossip syncers
-			// active.
-			for _, syncer := range syncerPeers {
-				syncer.FilterGossipMsgs(announcementBatch...)
-			}
+			splitAnnouncementBatch := splitAnnouncementBatches(
```

Comment:
I agree the function signature is simpler, but then on the other hand it's less clear what inputs are being used to calculate `subBatchSize`. 
Testing is also simpler my way :)

Reply:
up to you though, was thinking it would help save from extensive wrapping. i'm fine with either, tho the function arguments should be wrapped per the style guidelines in either case

fwiw i think testing is about the same, as we'd only need to set/modify the relevant config options and invoke the appropriate method on the gossiper (no need to spin up a full gossiper)